File: crates/gen-guest-ts/tests/anon.ts
declare global {
	interface Window {
		__TAURI_INVOKE__<T>(
			cmd: string,
			args?: Record<string, unknown>,
		): Promise<T>;
	}
}
const invoke = window.__TAURI_INVOKE__;
export type Error = "success" | "failure";
export async function optionTest(): Promise<string | null> {
	const result = await invoke<string | null>(
		"plugin:9f005dd416978e86|option-test",
	);
	return result;
}

File: crates/gen-guest-rust/tests/anon.rs
#[allow(clippy::all, unused)]
pub mod anon {
    #[repr(u8)]
    #[derive(Debug, Clone, Copy, PartialEq, ::serde::Deserialize)]
    pub enum Error {
        Success,
        Failure,
    }
    pub async fn option_test() -> Result<Option<String>, Error> {
        ::tauri_bindgen_guest_rust::invoke("plugin:9f005dd416978e86|option-test", ())
            .await
            .unwrap()
    }
}
